chart/templates/preflight.yaml

```diff
@@ -0,0 +1,127 @@
+              - pass:
+                  when: connected == true
+                  message: The PostgreSQL server is available
+        - postgres:
+            checkName: PostgreSQL must be v11.x or later
+            collectorName: workspace-db
+            outcomes:
+              - fail:
+                  when: connected == false
+                  message: Cannot connect to PostgreSQL server
+              - fail:
+                  when: version < 11.x
+                  message: The PostgreSQL server must be at least version 11
+              - pass:
+                  message: The PostgreSQL verion checks out
+        - clusterVersion:
+            outcomes:
+              - fail:
+                  when: "< 1.20.0"
+                  message: The application requires Kubernetes 1.20.0 or later, and recommends 1.21.0 or later.
+                  uri: https://www.kubernetes.io
+              - warn:
+                  when: "< 1.21.0"
+                  message: Your cluster meets the minimum version of Kubernetes, but we recommend you update to 1.21.0 or later.
+                  uri: https://kubernetes.io
+              - pass:
+                  message: Your cluster meets the recommended and required versions of Kubernetes.
+        - containerRuntime:
+            outcomes:
+              - pass:
+                  when: "== containerd"
+                  message: containerd container runtime was found.
+              - fail:
+                  message: Did not find containerd container runtime.
+        - distribution:
+            outcomes:
+              - fail:
+                  when: "== docker-desktop"
+                  message: The application does not support Docker Desktop clusters.
+              - fail:
+                  when: "== microk8s"
+                  message: The application does not support MicroK8s clusters.
+              - fail:
+                  when: "== minikube"
+                  message: The application does not support minikube clusters.
+              - pass:
+                  when: "== eks"
+                  message: EKS is a supported distribution.
+              - pass:
+                  when: "== gke"
+                  message: GKE is a supported distribution.
+              - pass:
+                  when: "== aks"
+                  message: AKS is a supported distribution.
+              # Will be supported in the future
+              - pass:
+                  when: "== kurl"
+                  message: kURL is a supported distribution.
+              - pass:
+                  when: "== digitalocean"
+                  message: DigitalOcean is a supported distribution.
+              - warn:
+                  message: Unable to determine the distribution of Kubernetes.
+        - nodeResources:
+            checkName: Total CPU Cores
+            outcomes:
+              - fail:
+                  when: "sum(cpuCapacity) < 2"
```

Comment:
Here we should define 5 as our doc say and recommend as 6 for example.
Based on our public doc https://docs.carto.com/carto-self-hosted/key-concepts/deployment-requirements#orchestrated-deployment-kubernetes